doc/en/thymio-api.rst

```diff
@@ -0,0 +1,503 @@
+Programming Interface
+=====================
+
+This page describes the programming capabilities of Thymio. It lists
+the different variables and functions and indicates to which elements
+of the robot they refer.
+
+Standard library
+----------------
+
+The Thymio comes with the Aseba `standard library of native functions,
+documented on its own page <aseba-std-natives.rst>`__.
+
+Buttons
+-------
+
+Thymio holds 5 capacitive buttons corresponding to the arrows and to the
+central button. Five variables hold the state of these buttons (1 =
+pressed, 0 = released):
+
+-  ``button.backward`` : backward arrow
+-  ``button.left`` : left arrow
+-  ``button.center`` : central button
+-  ``button.forward`` : forward arrow
+-  ``button.right`` : right arrow
+
+Thymio updates this array at a frequency of 20 Hz and generates the
+``buttons`` event after every update. Moreover, for each of these
+buttons, when it is pressed or released, a corresponding event with the
+same name is fired.
+
+Distance sensors
+----------------
+
+.. rubric:: Horizontal
+   :name: horizontal
+
+Thymio has 7 distance sensors around its periphery. An array of 7
+variables, ``prox.horizontal``, holds the values of these sensors:
+
+-  ``prox.horizontal[0]`` : front left
+-  ``prox.horizontal[1]`` : front middle-left
+-  ``prox.horizontal[2]`` : front middle
+-  ``prox.horizontal[3]`` : front middle-right
+-  ``prox.horizontal[4]`` : front right
+-  ``prox.horizontal[5]`` : back left
+-  ``prox.horizontal[6]`` : back right
+
+The values in this array vary from 0 (the robot does not see anything)
+to several thousand (the robot is very close to an obstacle). Thymio
+updates this array at a frequency of 10 Hz, and generates the ``prox``
+event after every update.
+
+.. rubric:: Ground
+   :name: ground
+
+Thymio holds 2 ground distance sensors. These sensors are located at the
+front of the robot. As black grounds appear like no ground at all (black
+absorbs the infrared light), these sensors can be used to follow a line
+on the ground. Three arrays hold the values of these sensors:
+
+-  ``prox.ground.ambiant`` : ambient light intensity at the ground,
+   varies between 0 (no light) and 1023 (maximum light)
+-  ``prox.ground.reflected`` : amount of light received when the sensor
+   emits infrared, varies between 0 (no reflected light) and 1023
+   (maximum reflected light)
+-  ``prox.ground.delta`` : difference between reflected light and
+   ambient light, linked to the distance and to the ground colour.
+
+For each array, the index 0 corresponds to the left sensor and the index
+1 to the right sensor. As with the distance sensors, Thymio updates this
+array at a frequency of 10 Hz, , and generates the (same) ``prox`` event
```

Comment:
double ", ,"

Reply:
As it should be http://docutils.sourceforge.net/docs/user/rst/quickref.html#inline-markup